server/api/api_accessor.py

```diff
@@ -28,47 +28,45 @@ def __init__(self):
     def update_token(self, token):
         self.token = token
 
-    def __enter__(self):
-        if not self.session:
-            try:
-                self.session = OAuth2Session(
-                    client=self.client,
-                    token=self.token,
-                    auto_refresh_url=API_TOKEN_URI,
-                    auto_refresh_kwargs={
-                        "client_id": API_CLIENT_ID,
-                        "client_secret": API_CLIENT_SECRET
-                    },
-                    token_updater=self.update_token
-                )
-                self.token = self.session.fetch_token(
-                    token_url=API_TOKEN_URI,
-                    client_id=API_CLIENT_ID,
-                    client_secret=API_CLIENT_SECRET
-                )
-            except:
-                self.__exit__(*sys.exc_info())
-
-        return self.session
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        if exc_type == MissingTokenError:
```

Comment:
What's wrong with context manager?

Reply:
It is of no use here. Context managers are good when you allocate a resource and want to make sure you clean it up when you're done, but here we're creating a resource then keeping it around forever. Grabbing it in a context manager then making special checks to ensure it doesn't do what it was designed to do is pointless, especially since a try-catch block does _exactly_ what we want to do.